test/extended/prometheus/prometheus.go

```diff
@@ -49,6 +53,120 @@ type TelemeterClientConfig struct {
+			namespacesToSkip.Delete(componentNamespace)
+		}
+	})
+
+	g.It("should not be accessible without authorization [Serial]", func() {
+		var errs []error
+
+		g.By("verifying all targets returns 401 or 403 or 500 without authorization")
+		execPod := exutil.CreateExecPodOrFail(oc.AdminKubeClient(), oc.Namespace(), "execpod-targets-authorization")
+		defer func() {
+			err := oc.AdminKubeClient().CoreV1().Pods(execPod.Namespace).Delete(context.Background(), execPod.Name, *metav1.NewDeleteOptions(1))
+			o.Expect(err).NotTo(o.HaveOccurred(), fmt.Sprintf("Delete pod %s/%s", execPod.Namespace, execPod.Name))
+		}()
+
+		contents, err := helper.GetURLWithToken(helper.MustJoinUrlPath(prometheusURL, "api/v1/targets"), bearerToken)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		targets := &prometheusTargets{}
+		err = json.Unmarshal([]byte(contents), targets)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		for _, target := range targets.Data.ActiveTargets {
+			ns := target.Labels["namespace"]
+			if !strings.HasPrefix(ns, "openshift-") {
+				continue
+			}
+			if componentNamespace != "" && ns != componentNamespace {
+				continue
+			}
+			pod := target.Labels["pod"]
+			e2e.Logf("Checking via pod exec status code from the scaple url %s for pod %s/%s without authorization (skip=%t)", target.ScrapeUrl, ns, pod, namespacesToSkip.Has(ns))
+			// Error out only when the status code is obtained
+			// Error on executing the cmd is skipped here because the endpoint does not expose any information.
+			// Promethus may have troubles to scrape it but that is out of the scape of this test
+			// The same reasoning holds for status code 500
+			err = helper.ExpectURLStatusCodeExecViaPod(execPod.Namespace, execPod.Name, target.ScrapeUrl, true, http.StatusUnauthorized, http.StatusForbidden, http.StatusInternalServerError)
+			e2e.Logf("The scaple url %s for pod %s/%s without authorization returned error %v (skip=%t)", target.ScrapeUrl, ns, pod, err, namespacesToSkip.Has(ns))
+			if err != nil && !namespacesToSkip.Has(ns) {
+				errs = append(errs, fmt.Errorf("the scaple url %s for pod %s/%s is accessible without authorization: %w", target.ScrapeUrl, ns, pod, err))
+			}
+		}
+
+		g.By("verifying all service monitors are configured with authorization")
+		client, err := prometheusoperatorv1client.NewForConfig(oc.AdminConfig())
+		o.Expect(err).NotTo(o.HaveOccurred(), "Create monitoring client")
+
+		serviceMonitorList, err := client.ServiceMonitors("").List(context.Background(), metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred(), "List service monitors")
+
+		for _, sm := range serviceMonitorList.Items {
+			// we do not check service monitors for user-workload-monitoring
+			if !strings.HasPrefix(sm.Namespace, "openshift-") {
+				continue
+			}
+			if componentNamespace != "" && sm.Namespace != componentNamespace {
+				continue
+			}
+			ok := authorizationConfigured(sm)
```

Comment:
I think we can do without this check, the curl is enough;
- setting auth via ServiceMonitor doesn't mean it's used.
- cone fewer thing to maintain.
- Maybe the auth is set somewhere else... (Note that the test doesn't take into account podMonitors etc)

Reply:
Removed.

> Maybe the auth is set somewhere else... (Note that the test doesn't take into account podMonitors etc)

For my curioctiy: Is it possible that for some SM it failed `authorizationConfigured(sm)` but it's target is protected?

The `Note` part gives me the feeling that the protection is in some `podMonitor`. Could you elaborate it?